core/src/main/java/org/jobrunr/server/BackgroundJobPerformer.java

```diff
@@ -144,4 +147,18 @@ private boolean isJobServerStopped(Exception e) {
     private boolean isJobNotFoundException(Exception e) {
         return e instanceof JobNotFoundException;
     }
+
+    /**
+     * JobRunr uses reflection to run jobs. Any error in jobs is wrapped in {@link InvocationTargetException}.
+     * Job details shows {@link InvocationTargetException} and its stacktrace on UI
+     * with lots of internal details not related to the job.
+     * It makes harder for users to read exceptions
+     * and leaves less space for the actual errors' stacktraces on UI.
+     */
+    @VisibleFor("testing")
+    static Throwable unwrapException(Exception e) {
+        if (!(e instanceof InvocationTargetException)) return e;
```

Comment:
Let's change this to:

```java
if (e instanceof InvocationTargetException && e.getCause() instanceof Exception) return e.getCause();

return e;
```

FYI `e.getTargetException()` is the legacy way. The javadocs recommend to use `e.getCause()`.

Reply:
nice, it looks better this way. thank you.